### DISCUSSION TITLE
Homebrew immediately uninstalling openssl every time I install it

### DISCUSSION THREAD

**Discussion by numbermaniac** [Homebrew/discussions#1260] [Getting started] [answered]
Every time I try to `brew install openssl@1.1`, Homebrew is deleting it immediately after installing it.

```sh
==> Summary
🍺  /usr/local/Cellar/openssl@1.1/1.1.1k: 8,071 files, 18.4MB
Removing: /usr/local/Cellar/openssl@1.1/1.1.1k... (8,071 files, 18.4MB)
```

How do I make this stop happening? I want it to stay installed.

**Comment by carlocab** [comment]
Try running
```bash
HOMEBREW_NO_INSTALL_CLEANUP=1 brew install openssl@1.1
```

**Reply by numbermaniac** [reply]
Looks like that worked, thanks. However, if I do `brew cleanup` manually afterwards, Homebrew is deleting openssl again. That shouldn't be happening, right?

**Reply by carlocab** [reply]
No, it shouldn't. What's the output of `brew config` and `brew doctor`?

**Reply by numbermaniac** [reply]
It's fairly long, so I've put the outputs of both those commands into a file and uploaded it here: https://mystb.in/GenresRoofReleased.txt

**Reply by SMillerDev** [reply]
> Warning: A newer Command Line Tools release is available.
Update them from Software Update in System Preferences or run:
  softwareupdate --all --install --force

> If that doesn't show you any updates, run:
  sudo rm -rf /Library/Developer/CommandLineTools
  sudo xcode-select --install

> Alternatively, manually download them from:
  https://developer.apple.com/download/more/.

---

> You have uncommitted modifications to Homebrew/homebrew-core.
If this is a surprise to you, then you should stash these modifications.
Stashing returns Homebrew to a pristine state but can be undone
should you later need to do so for some reason.
  cd /usr/local/Homebrew/Library/Taps/homebrew/homebrew-core && git stash && git clean -d -f

Could you try these two suggestions from brew doctor?

**Reply by carlocab** [reply]
Try doing
```
brew update-reset
```
but note the following:
```
❯ brew help update-reset
Usage: brew update-reset [repository ...]

Fetch and reset Homebrew and all tap repositories (or any specified
repository) using git(1) to their latest origin/HEAD.

Note: this will destroy all your uncommitted or committed changes.
```
Also, try updating your CLT to version 11.3.1.

**Reply by numbermaniac** [reply]
I tried all three of these and then restarted my computer - unfortunately openssl is still being removed during cleanup. This is very weird - I've never seen this happen before.

**Reply by carlocab** [reply]
That is very weird. What's the output of `brew config` and `brew doctor` now?

**Comment by carlocab** [comment, marked as answer]
It looks like this is a bug in `brew`. Fix in progress: https://github.com/Homebrew/brew/pull/11114

**Reply by numbermaniac** [reply]
After updating to the newly-released Homebrew 3.1.1 and then re-installing openssl, `brew cleanup` is no longer removing it. Thank you for fixing it so quickly! :)
